File: 2013-06-07/python/exercise2.py
from pyplasm import *
import random

mtrx = []


def RNDM (x,y):
	f = (COS(x)+SIN(y))*random.random()
	m = [x,y,f]
	mtrx.append(m)
	return f 


dom1 = INTERVALS(20)(60)
dom2 = INTERVALS(40)(60)
domain = PROD([dom1,dom2])

def mountain (v):
	a = v[0]
	b = v[1]
	return [a, b, RNDM(a,b)]


model = MAP(mountain)(domain)
mountains = COLOR([0.6,0.4,0.2])(model)


lake = COLOR([0,0.58,0.71])(CUBOID([4,5,1]))

lake1 = T([1,2,3])([1,8.5,-1])(lake)
lake2 = T([1,2,3])([7.3,27,-1])(lake)

VIEW(STRUCT([lake1,lake2,mountains]))

File: 2013-06-07/javascript/exercise2.js
var mtrx = new Array();


var RNDM = function(x,y){
var f = (COS(x)+SIN(y))*Math.random();
var m = [x,y,f];
mtrx.push(m);
return f; 
};

var search_m = function(x,y,z){
	var z = null
	for (var i = 0; i <= mtrx.length-1; i++) {
		var c = mtrx[i];
		if ((c[0]==x)&&(c[1]==y))
			z = c[3];
	};
	return z
}


var dom = DOMAIN([[0,20],[0,40]])([60,60]);
var mountain = function (k) {
return function (v) {
var a = v[0];
var b = v[1];
return [a, b, RNDM(a,b)];
};
};
var mapping = mountain(3);
var model = MAP(mapping)(dom);
var DTM = COLOR([205/255,133/255,63/255])(model);

//ES2 DRAW ONE OR MORE LAKES ON THE MOUNTAINS MODEL
var lake = COLOR([0/255,149/255,182/255,0.9])(CUBOID([4,5,1]));

var lake1 = T([0,1,2])([1,8.5,-1])(lake);
var lake2 = T([0,1,2])([7.3,27,-1])(lake);

DRAW(STRUCT([lake1,lake2,DTM]));